packages/web/src/game/scenes/Game.ts

```diff
@@ -548,18 +533,34 @@ export default class GameScene extends Scene {
       },
     );
     networkHandler.on(NetworkEvents.TICK, (data: DataTypes[NetworkEvents.TICK]) => {
-      const currentMap = this._mapHelper.loadedMaps[this._player.currentMap]; 
-      // sine function imitating day/night cycle
       // only works with 1440 minutes a day cycle
-      // TODO: defined in ldtk? along with position of sun
       const cursor = 230;
-      const maxSunIntensity = 300; 
-      const maxMoonIntensity = 50;
-      const fn = (Math.sin((data.time / cursor) - (Math.PI/2)) + 1) / 2;
-      const color = this.dayColor.map(color => (Math.round(color * fn)).toString(16)).join('');
-      this.lights.setAmbientColor(Number.parseInt(color, 16) < 272722 ? 
-        Number.parseInt(this.nightColor.map(color => (Math.round(color * (0.2 - fn))).toString(16)).join(''), 16) : 
-          Number.parseInt(color, 16));
+
+      // Calculate the intensity factor based on the time of day
+      const minFactor = 0.35;
+      const maxFactor = 0.9;
+      const timeFactor = Math.min(Math.max((Math.sin((data.time / cursor) - (Math.PI/2)) + 1) / 2, minFactor), maxFactor);
+
+      // Calculate the color based on the current time factor
+      const dayColorHex = this.dayColor.map(color => (Math.round(color * timeFactor)).toString(16)).join('');
+      const nightColorHex = this.nightColor.map(color => (Math.round(color * (0.2 - timeFactor))).toString(16)).join('');
+
+      // Determine which color to use based on time of day
+      let ambientColor;
+
+      if (Number.parseInt(dayColorHex, 16) < 272722)
+      {
+      // nighttime
+      ambientColor = Number.parseInt(nightColorHex, 16);
```

Comment:
Please stay close to the code-style of the file (indentation, brackets etc.)

Reply:
my bad